message_processor.go

```diff
@@ -50,34 +87,50 @@ func consumeLog(lf *logPair) {
 			logrus.WithField("id", lf.info.ContainerID).WithError(err).Debug("Ignoring error")
 			dec = protoio.NewUint32DelimitedReader(lf.stream, binary.BigEndian, 1e6)
 		}
-		if sendMessage(lf.splunkl, &buf, lf.info.ContainerID) == false {
-			continue
+		// Apprend to partial buffer
+		appendToPartialBuffer(&pBuf, &buf)
+		// Check buffer hold duration
+		if isPartialBufferHoldDurationExpired(timer.C) && buf.Partial {
+			logrus.Debug("Force partial bit to false due to buffer hold duration expiry")
+			buf.Partial = false
 		}
-		if sendMessage(lf.jsonl, &buf, lf.info.ContainerID) == false {
+
+		if sendMessage(lf.splunkl, &buf, &pBuf, lf.info.ContainerID) == false {
 			continue
 		}
 
+		if sendMessage(lf.jsonl, &buf, &pBuf, lf.info.ContainerID) == false {
+			continue
+		}
+		//partial buffer reset
+		if pBuf.bufferReset {
+			pBuf.pmsg.Reset()
+		}
 		buf.Reset()
 	}
-
+	timer.Stop()
 }
 
 // send the log entry message to logger
-func sendMessage(l logger.Logger, buf *logdriver.LogEntry, containerid string) bool {
+func sendMessage(l logger.Logger, buf *logdriver.LogEntry, pBuffer *pmsgBuffer, containerid string) bool {
 	var msg logger.Message
 	if !shouldSendMessage(buf.Line) {
 		return false
 	}
+	if !buf.Partial || pBuffer.bufferMaximum <= pBuffer.pmsg.Len() {
```

Comment:
1. It appears to me it is not correct to return `true` when we don't need send ?
2. Seems we need merge this condition to `shouldSendMessage` ?

Reply:
Returning True does not reset the partial buffer only the original buffer and we don't need it anymore since we copy the msg to the partial buffer.

Reply:
@chenziliang Yes, I did merge shouldSendMessage so we don't add msg to temp partial buffer